src/v/kafka/server/replicated_partition.h

```diff
@@ -48,10 +55,21 @@ class replicated_partition final : public kafka::partition_proxy::impl {
     }
 
     model::offset high_watermark() const final {
+        if (
+          _partition->is_read_replica_mode_enabled()
+          && _partition->cloud_data_available()) {
+            return _partition->last_cloud_offset();
```

Comment:
Kafka high watermark semantics requires us to return `(last log offset + 1)`. Is that the case for `last_cloud_offset` or should we add it here ?

Reply:
fixed, added +1